Add profile onSubmit message and redirect logic

B77Mills opened the pull request:
On submit hide the form and display a success message.  The message will be one of the two following message depending on if there is a returnTo url set or not.

### With returnTo:
Your profile has been saved.
If you are not automatically redirected click here to continue.
<img width="1202" alt="Screen Shot 2022-11-11 at 10 15 00 AM" src="https://user-images.githubusercontent.com/3845869/201383700-07330031-38b6-498d-9585-368088aa4964.png">


### Without returnTo:
Your profile has been saved.
Click here to continue modiying your profile or click here to return to the home page.
<img width="1175" alt="Screen Shot 2022-11-11 at 10 14 38 AM" src="https://user-images.githubusercontent.com/3845869/201383723-a9c56a5f-d5ae-41f2-a400-393723d3692e.png">

B77Mills committed:
Add profile onSubmit message and redirect logic

On submit hide the form and display a success message.  The message will be one of the two following message depending on if there is a returnTo url set or not.

WIth returnTo:
Your profile has been saved.
If you are not automatically redirected click here to continue.

Without returnTo:
Your profile has been saved.
Click here to continue modiying your profile or click here to return to the home page.

B77Mills committed:
Add Monorail theme overrides for font styling.

solocommand commented on packages/marko-web-theme-monorail/scss/components/_identity-x.scss:
```diff
@@ -34,4 +34,15 @@
   }
+  &--profile {
+    .success-message {
+      color: pink;
```
Why pink? just use `text-success` class and allow the theme to dictate what that color actually is.

solocommand commented on packages/marko-web-identity-x/browser/profile.vue:
```diff
@@ -176,6 +176,20 @@
+        </div>
+        <div v-else class="success-message__message">
+          <a href="#" @click="handleReload()">Click here</a> to continue modiying your profile
+          or <a href="/">click here</a> to return to the home page.
```
Make this a button with `btn-link` class rather than an A tag with a search param. `modiying` should be `modifying`, and I'd recommend this text:
> [Click here]() to make further changes to your preferences, or [return to the home page]().

solocommand commented on packages/marko-web-identity-x/browser/profile.vue:
```diff
@@ -322,6 +336,10 @@ export default {
       type: String,
       default: null,
     },
+    returnToDelay: {
```
This must be defined and passed through in both the `marko` components, and any other components that include the profile (such as Authenticate, Login, and Comment)

B77Mills replied:
@solocommand  I did add this to marko.json and added passing it through the marko component if it is sent along with the input.   

As for adding it to the other things mentioned. The commenting & login component do not call or reference the profile component and the auth component is overriding the submit(profile-update) action & forcing a redirect already [line 27](https://github.com/parameter1/base-cms/blob/master/packages/marko-web-identity-x/browser/authenticate.vue)

I don't think this is currently being called in any other spots.

B77Mills replied:
@solocommand  I also checked the gated content part which should be using the reload-page-on-submit=true prop which would force the content page to reload on profile submit.  But in theory the user should be intercepted at the authentication step to finish filling out the required fields and then get redirected to the content landing page.  If they did some how get to the content page without a fully filled out profile page this would still reload the page on submission of a full profile.

solocommand commented on packages/marko-web-identity-x/browser/profile.vue:
```diff
@@ -176,6 +176,20 @@
+          Your profile has been saved.
+        </div>
+        <div v-if="returnTo" class="success-message__message">
+          If you are not automatically redirected <a :href="returnTo">click here</a> to continue.
```
> You will be automatically redirected in {{ returnToDelay / 1000 }} seconds. [Click here]() to continue.

solocommand reviewed (changes requested):
From the UX side, 5 seconds is a **long** time to wait after the form has been submitted. While I don't like the idea of the form hiding in the first place, making the user wait 5 seconds to continue (especially when this is only an intermediate step such as inline gated content or redirect to external for LME) seems frustrating.

Regarding the form collapse: This will incur a lot of motion, and may end up being _more_ frustrating for the user. The 'submitted' message will flash for a frame or so next to the submit button, then immediately disappear and scroll to the top of the page (due to the length of the form no longer pushing them down). This in turn will cause the footer to take up most of the screen.

The success message should be appended to the bottom of the form, and should not cause elements to disappear and jump around on the page.

B77Mills committed:
Remove testing color setting

B77Mills committed:
Define and pass returnToDelay prop

B77Mills committed:
update success message classes

B77Mills committed:
Revert classing to not use bootstrap text-success

B77Mills committed:
move reload above redirect

If the page is suppose to reload force that to happen before it tries to redirect.

B77Mills committed:
Fix linting spacing issue with wrapping text

B77Mills committed:
Remove ref to submit message & add scroll logic on submit

Remove old submitMessage logic as it will never show when the wrapping element is now in a v-if=“!didSubmit” check and the new messaging is moved to the else portion of that.

I also added logic to scroll to top of the page on submit to prevent the user from being shown the footer when the form hides.

B77Mills committed:
reduce delay to 3 seconds and update message to include time

B77Mills committed:
Fix odd extra space in click here link text

B77Mills committed:
Use a button instead of a span with correct classing.

B77Mills committed:
Add type=“button”

B77Mills commented:
![](https://github.trello.services/images/mini-trello-icon.png) [On profile update, redirect to homepage if no redirect URL is present](https://trello.com/c/4Y2LZzWZ/105-on-profile-update-redirect-to-homepage-if-no-redirect-url-is-present)
